gateways/clients/hathor_core_client.py

```diff
@@ -64,6 +64,35 @@ async def get(
             self.log.error("hathor_core_error", path=path, error=repr(e))
             callback({"error": repr(e)})
 
+    async def post(
```

Comment:
While it is not used by push_tx I think we should support the same methods on both async and sync clients so I created the async post as well